docs/adr/adr-0017-finding-accessible-colors-and-creating-design-tokens.md

```diff
@@ -0,0 +1,168 @@
+
+We know we may need to **add more shades or colors later on but we want to only select and add them when needed**.
+
+### Create Design Tokens
+
+The Design Token concept comes from the Salesforce team and has been defined by them as follows:
+> Design tokens are the visual design atoms of the design system — specifically, they are named entities that store visual design attributes. We use them in place of hard-coded values (such as hex values for color or pixel values for spacing) in order to maintain a scalable and consistent visual system for UI development.
+
+Source: [salesforce - Design Tokens](https://www.lightningdesignsystem.com/design-tokens/).
+
+Our work on Design Tokens is based on two articles:
+* [Design Tokens - What are they & how will they help you? by Lukas Oppermann](https://lukasoppermann.medium.com/design-tokens-what-are-they-how-will-they-help-you-b73f80f602ab)
+* [Naming Tokens in Design System by Nathan Curtis](https://medium.com/eightshapes-llc/naming-tokens-in-design-systems-9e86c7444676)
+
+These articles helped us grasp the concept of Design Tokens and find an architecture / strategy that would fit our needs.
+
+NOTE: Color Design Tokens are only the first step, see "What's next" section.
+
+#### No abstraction level
+
+Most of the examples we came accross insisted on building Design Tokens that were **Technology Agnostic and relied on tools to transform tokens into variables** that could be consumed with each specific technology (Web, Native mobile platforms, etc.).
+
+This made sense for very big projects distributed accross several different platforms but our project is a little bit different:
+* Our team is only made of developers and we don't use any specific design tool,
+* Our components are only for the Web and as close as possible to the standards.
+
+For these reasons, we chose to **only rely on CSS variables and avoid using additional tools** like [Style Dictionary](https://amzn.github.io/style-dictionary/#/) that would only have added more complexity with no actual benefits.
+If necessary, this is something we could easily reconsider later on.
+
+#### Variable naming & architecture
+
+When creating our variables, we tried to think as much as possible about Developer Experience, theming and customisation.
+
+The [Design Tokens - What are they & how will they help you?](https://lukasoppermann.medium.com/design-tokens-what-are-they-how-will-they-help-you-b73f80f602ab) article helped us a lot to build a system that could scale and allow for customisation later on.
+In this article, the author distinguishes two kind of tokens:
+* "choices" - our colors.
+For instance:
+```css
+:root {
+  --color-black: #262626;
+}
+```
+* "decisions" - implementations of our colors.
+For instance:
+```css
+:root {
+  --color-text-default: var(--color-black);
+}
```

Comment:
unfortunately even with a selector like `.myclass`, the syntax highlight did not work (it only highlights known CSS property names like `color`, etc.)